scopes/core.md

```diff
@@ -0,0 +1,176 @@
+
+If no network scope is assigned, the platform MUST join the application to a default network. Within that default network, all components in the application configuration MUST be able to communicate to each other, and health probes MUST be able to contact any components who define health checking rules. The network joined, however, is platform-dependent. For example, cluster-based environments (such as Kubernetes) declare cluster-wide networks. Conversely, a truly serverless implementation may join the components to a network that only includes the components and the health checking probes.
+
+### Schematic
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `networkId` | `string` | Y | | The id of the network, e.g. vpc-id, VNet name. |
+| `subnetIds` | `array` | Y | | A comma separated list of IDs of the subnets within the network. For example, 'vsw-123' or 'vsw-123','vsw-456'. There could be more than one subnet because there is a limit in the number of IPs in a subnet. If IPs are taken up, operators need to add another subnet into this network. |
+| `internetGatewayType` | `string` | Y | | The type of the gateway, options are 'public', 'nat'. Empty string means no gateway. |
+
+### Spec
+
+The network scope is defined by the following spec:
+
+```yaml
+apiVersion: core.oam.dev/v1alpha1
+kind: ScopeDefinition
+metadata:
+  name: networkscope.core.oam.dev
+spec:
+  allowComponentOverlap: true
+  definitionRef:
+     name: networkscope.core.oam.dev
+```
+
+OAM runtime should store the schema with a version, the schema SHOULD be able to validate. For example using JSON Schema like this:
+
+```json
+    {
+      "$schema": "http://json-schema.org/draft-07/schema#",
+      "type": "object",
+      "required": [
+        "networkId",
+        "subnetIds",
+        "internetGatewayType"
+      ],
+      "properties": {
+        "networkId": {
+          "type": "string",
+          "description": "The id of the network, e.g. vpc-id, VNet name."
+        },
+        "subnetIds": {
+          "type": "array",
+          "description": "A comma separated list of IDs of the subnets within the network. For example, 'vsw-123' or 'vsw-123','vsw-456'. There could be more than one subnet because there is a limit in the number of IPs in a subnet. If IPs are taken up, operators need to add another subnet into this network.",
+          "items": {
+             "type": "string"
+          }
+        },
+        "internetGatewayType": {
+          "type": "string",
+          "description": "The type of the gateway, options are 'public', 'nat'. Empty string means no gateway."
+        }
+      }
+    }
+```
+
+### Instance Example
+
+For example, a network scope instance can be like below, asuming the version is `v1alpha1`:
+
+```yaml
```

Comment:
What does this instance mean? We only embedded it in the applicationConfig now, right?

Reply:
Not for scopes; they're distinct instances that are referenced by an `ApplicationConfiguration`. Traits are the only thing that gets embedded directly in an `ApplicationConfiguration`.